.github/workflows/coverage-report.yml

```
@@ -30,12 +30,12 @@ jobs:
       name: github-pages
       url: ${{ steps.deployment.outputs.page_url }}
     runs-on: ubuntu-latest
-    if: ${{ !github.event.pull_request.draft }}
+    if: ${{ !github.event.pull_request.draft }} && ${{ github.base_ref == 'refs/heads/main' }}
```

actionlint

if: condition "${{ !github.event.pull_request.draft }} && ${{ github.base_ref == 'refs/heads/main' }}" is always evaluated to true because extra characters are around ${{ }}

(if-cond)